Hestia/app/src/androidTest/java/com/rugged/application/hestia/backend/models/ActivatorStateTest.java

```diff
@@ -1,66 +1,143 @@
+        Float newFloatRawState = 0.3f;
+        assertNotEquals(DEFAULT_FLOAT_RAW_STATE, newFloatRawState);
         floatActivatorState.setRawState(newFloatRawState);
+        assertEquals(newFloatRawState, floatActivatorState.getRawState());
 
-        assertEquals(newBoolRawState,boolActivatorState.getRawState());
-        assertEquals(newFloatRawState,floatActivatorState.getRawState());
-        assertNotEquals(DEFAULT_BOOL_RAW_STATE,boolActivatorState.getRawState());
-        assertNotEquals(DEFAULT_FLOAT_RAW_STATE,floatActivatorState.getRawState());
+        // Test string activator state
+        assertEquals(DEFAULT_STRING_RAW_STATE, stringActivatorState.getRawState());
+        String newStringRawState = "newState";
+        assertNotEquals(DEFAULT_STRING_RAW_STATE, newStringRawState);
+        stringActivatorState.setRawState(newStringRawState);
+        assertEquals(newStringRawState, stringActivatorState.getRawState());
     }
 
     @Test
     public void setAndGetTypeTest() {
+        // Test bool activator state
         assertEquals(DEFAULT_BOOL_TYPE, boolActivatorState.getType());
-        assertEquals(DEFAULT_FLOAT_TYPE, floatActivatorState.getType());
-
         String newBoolType = "newBoolType";
-        String newFloatType = "newFloatType";
+        assertNotEquals(DEFAULT_BOOL_TYPE, newBoolType);
         boolActivatorState.setType(newBoolType);
+        assertEquals(newBoolType, boolActivatorState.getType());
+
+        // Test float activator state
+        assertEquals(DEFAULT_FLOAT_TYPE, floatActivatorState.getType());
+        String newFloatType = "newFloatType";
+        assertNotEquals(DEFAULT_FLOAT_TYPE, newFloatType);
         floatActivatorState.setType(newFloatType);
+        assertEquals(newFloatType, floatActivatorState.getType());
+
+        // Test string activator state
+        assertEquals(DEFAULT_STRING_TYPE, stringActivatorState.getType());
+        String newStringType = "newStringType";
+        assertNotEquals(DEFAULT_STRING_TYPE, newStringType);
+        stringActivatorState.setType(newStringType);
+        assertEquals(newStringType, stringActivatorState.getType());
+    }
+
+    @Test
+    public void getRawStateJSONTest() {
+        // Test bool activator state
+        JsonPrimitive boolPrimitive = new JsonPrimitive(DEFAULT_BOOL_RAW_STATE);
+        assertNotNull(boolPrimitive);
+        assertEquals(boolPrimitive, boolActivatorState.getRawStateJSON());
+
+        // Test float activator state
+        JsonPrimitive floatPrimitive = new JsonPrimitive(DEFAULT_FLOAT_RAW_STATE);
+        assertNotNull(floatPrimitive);
+        assertEquals(floatPrimitive, floatActivatorState.getRawStateJSON());
+
+        // Test string activator state
+        JsonPrimitive stringPrimitive = new JsonPrimitive(DEFAULT_STRING_RAW_STATE);
+        assertNotNull(stringPrimitive);
+        assertEquals(stringPrimitive, stringActivatorState.getRawStateJSON());
+    }
```

Comment:
Instead of indication the different sections it would be better to divide them evenly.